okta/provider.go

```diff
@@ -179,6 +179,12 @@ func Provider() *schema.Provider {
 				DefaultFunc: schema.EnvDefaultFunc("OKTA_BASE_URL", "okta.com"),
 				Description: "The Okta url. (Use 'oktapreview.com' for Okta testing)",
 			},
+			"custom_endpoint": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("OKTA_CUSTOM_ENDPOINT", ""),
+				Description: "Custom URL endpoint for unit testing and caching",
+			},
```

Comment:
but I'm not seeing the value of having a `OKTA_CUSTOM_ENDPOINT`. The hostname of the org is already set by `OKTA_ORG_NAME` combined with `OKTA_BASE_URL`.

I think it would be more flexible if the env var that triggered all of this off was `OKTA_HTTP_PROTOCOL`.

For example, if `OKTA_HTTP_PROTOCOL=http`, that would be thing that is used to set up the `disableHTTPS` in `okta/config.go` and the old code `okta.WithOrgUrl(fmt.Sprintf("https://%v.%v", c.orgName, c.domain)),` would take `disableHTTPS` into consideration.